### PR TITLE
Add a dependency on dune-configurator for all packages that use dune.configurator

### PR DESCRIPTION
**Pull request by diml**
We are planning to split `dune.configurator` as a separate `dune-configurator` package in Dune 2.0.0. The `dune.configurator` name will still exist for backward compatibility reason, however packages that use it must declare an additional dependency on `dune-configurator` in order to be compatible with Dune >= 2.0.0.

In preparation of this, this PR adds an empty `dune-configurator.1.0.0` that depends on Dune < 2.0.0 and adds it as a dependency of all packages that use `dune.configurator`.

### CODE REVIEW

**Comment by diml** [comment]
The failures are unrelated.

File: packages/fftw3/fftw3.0.8.2/opam [hunk 1]
```diff
@@ -14,6 +14,7 @@ build: [
     "env"
     "FFTW3_CFLAGS=-I /usr/local/include -L /usr/local/lib -I /usr/local/include -L /usr/local/lib"
     "dune"
+  "dune-configurator"
```
Comment (kit-ty-kate) [inline packages/fftw3/fftw3.0.8.2/opam]:
```suggestion
```
This shouldn't be there

Comment (diml) [reply packages/fftw3/fftw3.0.8.2/opam]: Actually, I believe it should:

```
$ opam source fft3.0.8.2
[fftw3.0.8.2] downloaded from cache at https://opam.ocaml.org/cache
Successfully extracted to /home/dim/code/fftw3.0.8.2
$ cd fftw3.0.8.2/
$ dune external-lib-deps @install
These are the external library dependencies in the default context:
- bigarray
- dune.configurator
- str
```

Comment (diml) [reply packages/fftw3/fftw3.0.8.2/opam]: Ah never-mind, I just realised that it was part of the build instructions...

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**